packages/contracts-bedrock/scripts/DeployImplementations.s.sol

```diff
@@ -265,6 +271,168 @@ contract DeployImplementationsOutput is BaseDeployIO {
         DeployUtils.assertValidContractAddress(address(_disputeGameFactoryImpl));
         return _disputeGameFactoryImpl;
     }
+
+    // -------- Deployment Assertions --------
+    function assertValidDeploy(DeployImplementationsInput _dii) public {
+        assertValidDelayedWETHImpl(_dii);
+        assertValidDisputeGameFactoryImpl(_dii);
+        assertValidL1CrossDomainMessengerImpl(_dii);
+        assertValidL1ERC721BridgeImpl(_dii);
+        assertValidL1StandardBridgeImpl(_dii);
+        assertValidMipsSingleton(_dii);
+        assertValidOpsmProxy(_dii);
+        assertValidOptimismMintableERC20FactoryImpl(_dii);
+        assertValidOptimismPortalImpl(_dii);
+        assertValidPreimageOracleSingleton(_dii);
+        assertValidSystemConfigImpl(_dii);
+    }
+
+    function assertValidOpsmProxy(DeployImplementationsInput _dii) internal {
+        // First we check the proxy as itself.
+        Proxy proxy = Proxy(payable(address(opsmProxy())));
+        vm.prank(address(0));
+        address admin = proxy.admin();
+        require(admin == address(_dii.superchainProxyAdmin()), "OPSM-10");
+
+        // Then we check the proxy as OPSM.
+        DeployUtils.assertInitialized({ _contractAddress: address(opsmProxy()), _slot: 0, _offset: 0 });
+        require(address(opsmProxy().superchainConfig()) == address(_dii.superchainConfigProxy()), "OPSM-20");
+        require(address(opsmProxy().protocolVersions()) == address(_dii.protocolVersionsProxy()), "OPSM-30");
+        require(LibString.eq(opsmProxy().latestRelease(), _dii.release()), "OPSM-50"); // Initial release is latest.
+
+        // Lastly we check it's implementation.
+        vm.prank(address(0));
+        OPStackManager impl = OPStackManager(proxy.implementation());
+        DeployUtils.assertInitialized({ _contractAddress: address(impl), _slot: 0, _offset: 0 });
+        require(address(impl.superchainConfig()) == address(_dii.superchainConfigProxy()), "OPSM-60");
+        require(address(impl.protocolVersions()) == address(_dii.protocolVersionsProxy()), "OPSM-70");
```

Comment:
Nit: it would feel slightly cleaner to me to put this be in a separate `assertValidOpsmImpl()` function.